client/src/partials/templates/modals/dateInterval.js

```diff
@@ -0,0 +1,69 @@
+angular.module('bhima.controllers')
+.controller('DateIntervalModalController', DateIntervalModalController);
+
+DateIntervalModalController.$inject = ['$uibModalInstance'];
+
+/**
+ * Date Interval Modal
+ * This controller permit to select Interval of date
+ * it returns the result as dateFrom and dateTo
+ */
+function DateIntervalModalController (Instance) {
+  var vm = this;
+
+  vm.options = [
+    { label : 'FORM.LABELS.TODAY', fn : day },
+    { label : 'FORM.LABELS.THIS_WEEK', fn : week },
+    { label : 'FORM.LABELS.THIS_MONTH', fn : month },
+    { label : 'FORM.LABELS.OTHER', fn : other },
+  ];
+
+  // expose to the viewe
+  vm.search = search;
+  vm.submit = submit;
+  vm.close  = close;
+
+  // start up the modal
+  startup();
+
+  // submit
+  function submit() {
+    Instance.close({ dateFrom: vm.dateFrom, dateTo: vm.dateTo });
+  }
+
+  // close
+  function close() {
+    Instance.dismiss();
+  }
+
+  function search (selection) {
+    vm.selected = selection.label;
+    selection.fn();
+  }
+
+  function day () {
+    vm.dateFrom = new Date();
+    vm.dateTo = new Date();
+  }
+
+  function week () {
```

Comment:
There is a service for this.  It's called `DateService`.  Could you use it? That way we only have to test a single service in the future.

This function would like this:
```js
// Assuming DateService is imported as Dates
function week() {
  vm.dateFrom = Dates.current.day();
  vm.dateTo = Dates.current.week();
}
```

Could you do the rest like this?